src/raft/mod.rs

```diff
@@ -37,7 +37,7 @@ mod read_only;
 
 pub use self::storage::{RaftState, Storage};
 pub use self::errors::{Result, Error, StorageError};
-pub use self::raft::{Raft, StateRole, Config, INVALID_ID, SoftState};
+pub use self::raft::{Raft, StateRole, Config, INVALID_ID, INVALID_INDEX, SoftState};
```

Comment:
What if we want to use INVALID_INDEX for first_index and last_index? I think `INVALID_INDEX` is ok, commit_index is one type of index, INVALID_INDEX is also an invalid commit index.